### PR TITLE
feat: add monitoring demo

### PR DESCRIPTION
**Pull request by n063h**
ref to #297

Provide a whole monitoring demo with Prometheus and Grafana.

# Usage

`cd examples/monitor` and run `docker-compose up`
Then you can open [Grafana](http://localhost:3000/d/4vV0bzE4z/mosec?orgId=1) to see the mosec dashboard
image

or open [Prometheus](http://127.0.0.1:9090) to query the metrics collected in mosec.
image

### CODE REVIEW

File: examples/python_side_metrics.py [hunk 1]
```diff
@@ -39,7 +39,12 @@
 
 metric_registry = CollectorRegistry()
 multiprocess.MultiProcessCollector(metric_registry)
-counter = Counter("inference_result", "statistic of result", ("status", "worker_id"))
+counter = Counter(
+    "inference_result",
+    "statistic of result",
+    ("status", "worker_id"),
+    registry=metric_registry,
```
Comment (n063h) [inline examples/python_side_metrics.py:46]: If the registry parameter is not specified, the Counter will not be monitored by Prometheus.

File: examples/monitor/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  mosec:
+    image: python:3.9-slim-buster
+    container_name: mosec
+    ports:
+      - "8000:8000"
+      - "5000:5000"
+    volumes:
+      - ../python_side_metrics.py:/root/python_side_metrics.py
+    entrypoint: ["/bin/bash", "-c","pip install -i https://pypi.tuna.tsinghua.edu.cn/simple -U mosec prometheus_client && python /root/python_side_metrics.py"]
```
Comment (kemingy) [inline examples/monitor/docker-compose.yml:13]: Don't do this. You can build from a local Dockerfile.

BTW, remove the mirror after testing.

File: examples/monitor/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    restart: always
```
Comment (kemingy) [inline examples/monitor/docker-compose.yml:14]: Should shutdown on failure for mosec.

File: examples/monitor/readme.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Start the monitoring system by running the following command:
+```bash
+docker-compose up -d
+```
+This command will start three containers: Mosec, Prometheus, and Grafana.
+
+
+# Accessing Prometheus
+Prometheus is a monitoring and alerting system that collects metrics from Mosec. You can access the Prometheus UI by visiting http://localhost:9090 in your web browser.
```
Comment (kemingy) [inline examples/monitor/readme.md:26]:
```suggestion
Prometheus is a monitoring and alerting system that collects metrics from Mosec. You can access the Prometheus UI by visiting http://127.0.0.1:9090 in your web browser.
```

`localhost` may be resolved to another IP.

File: examples/monitor/readme.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+# Accessing Grafana
+Grafana is a visualization tool for monitoring and analyzing metrics. You can access the Grafana UI by visiting http://localhost:3000 in your web browser. The default username and password are both admin.
```
Comment (kemingy) [inline examples/monitor/readme.md:29]: Same for this address.

**Review by kemingy — APPROVED** [review]

**Pull request merged**